hunts/templates/all_puzzles.html

```diff
@@ -323,11 +324,10 @@ <h5 class="modal-title" id="addtitle">Add Puzzle</h5>
             $(row).attr('id', `puzzle-${data[pk]}`);
             initDeleteTagHandlers(data);
         },
-        'drawCallback': function() {
-            $('#puzzles').on('draw.dt', initTreegrid);
-            $('#puzzles').on('search.dt', maybeExpandAll);
-        },
         'initComplete': function() {
+            initTreegrid();
+            $('#puzzles').on('search.dt', onSearch);
+            $('#puzzles').on('draw.dt', initTreegrid);
             reload();
         },
```

Comment:
I remember needing drawCallback so when the data got re-polled every 5-10 seconds, the table would automatically refresh. After your changes, if you enter an answer or add a puzzle in a different browser tab, will the table in the first tab still automatically refresh/redraw?

Reply:
Yep, still works. Every draw calls `drawCallback` and fires the `draw.dt` event. We just want exactly one listener to this event (the `$('#puzzles').on('draw.dt', initTreegrid)` call), but the existing code adds a new additional listener on every draw (so after n draws, there are n listeners).

Reply:
Nice! Glad someone understand JS better than I do :D